lib/ticker.dart

```diff
@@ -68,48 +68,38 @@ mixin Tick {}
 
 @immutable
 class ImmutableTick extends BlocState with Tick {
-  const ImmutableTick();
-  @override
-  int get hashCode => 0;
-  @override
-  bool operator ==(Object obj) => obj is Tick;
+  const ImmutableTick(TickType tickType) : super(tickType);
 }
 
 @immutable
 class TickTypeInstalled extends BlocState {
-  final TickType tickType;
-  const TickTypeInstalled(this.tickType);
+  const TickTypeInstalled(TickType tickType) : super(tickType);
 }
 
 class MutableTick extends BlocState with Tick {
-  MutableTick();
+  MutableTick(TickType tickType) : super(tickType);
 }
 
 enum TickType {
   mutable,
   immutable,
 }
 
-
 /* BLoC */
 
 ///
 class Ticker extends Bloc<BlocEvent, BlocState> {
   @override
-  BlocState get initialState => const Stopped();
+  BlocState get initialState => const Stopped(TickType.mutable);
 
-  Stream<State> whereState<State>() => whereType<State>();
+  StreamSubscription<int> _streamSubscription;
 
-  TickType _tickType = TickType.mutable;
```

Comment:
I would not recommend maintaining state within the bloc. It's best for all state to be part of the `BlocState`.